### DISCUSSION TITLE
Is the lack of the `Conditional` attribute intentional?

### DISCUSSION THREAD

**Discussion by stenci** [wtertinek/Linq2Acad#38] [Q&A] [answered]
I was expecting for `[Conditional("DEBUG")]` to be used on all the methods in `Helpers/Require.cs`.

Is there a good reason not to use it?

**Comment by wtertinek** [comment, marked as answer]
Hi, the methods in the `Require` class are a collection of guards that are used throughout the public API. It is intended that they are used not only in Debug mode, because they primarily test and report invalid user input (as early as possible).

**Reply by stenci** [reply]
> It is intended that they are used not only in Debug mode

It makes sense. They do look very similar to asserts, but if they were intended to be used only in Debug mode, they would be `Debug.AssertSomething` instead of `Require.Something`.
